app/controllers/concerns/reauthentication_required_concern.rb

```diff
@@ -20,13 +21,17 @@ def recently_authenticated_2fa?
   private
 
   def prompt_for_second_factor
-    store_location(request.url)
+    store_location(request.path)
     user_session[:context] = 'reauthentication'
 
     redirect_to login_two_factor_options_path
   end
 
   def store_location(url)
-    user_session[:stored_location] = url
+    if url === backup_code_continue_path
+      user_session[:stored_location] = account_path
+    else
+      user_session[:stored_location] = url
+    end
```

Comment:
@aduth Is this workaround appropriate to solve the backup code setup reauth bug? What happens currently is supposing a user is setting up 2 mfa and times-out before continuing after setting backup codes, the stored url '/backup_code_continue' gets passed and sent as a GET request by way of a redirect but it will 404 because there's no GET in the routes for that - [just patch](https://github.com/18F/identity-idp/blob/main/config/routes.rb#L282-L290)

 "PATCH /login/two_factor/webauthn HTTP/1.1" 302 - 0.0229
 "GET /backup_code_continue HTTP/1.1" 404 878 0.0437

Reply:
So is the idea that we send the user back to the account page if we're unable to maintain the original request method when redirecting them back? (`PATCH` in the case of backup codes)

If so, a follow-up question I'd have is: How does this work if a user is forced to reauthenticate before setting up their very first MFA in account creation? I'm sorta expecting that the account page may redirect them back to the MFA selection which could be okay.

One request I'd make is that we avoid hard-coding backup code considerations in this file. Maybe we can generalize this to set the stored location to `account_path` if `request.method != :get` ?

If we did go this route, I think a good follow-up task could revisit to see if we can update `stored_location` to properly support cases where the request method is non-GET, maybe using an approach similar to what we discussed when pairing (using an auto-submitting form similar to [SAML redirection](https://github.com/18F/identity-idp/blob/fab35dc2accf4f34fbe194baf553abea8e3384d4/app/views/saml_idp/shared/saml_post_binding.html.erb#L23-L29)).